### ISSUE TITLE
publishResult does not work - should use publisher

### ISSUE THREAD

**Issue by mooose** [GoogleCloudPlatform/nodejs-docs-samples#557]
Instead of 
```
function publishResult (topicName, data) {
  return pubsub.topic(topicName).get({ autoCreate: true })
    .then(([topic]) => topic.publish(data));
}
```

the following function works:
```
function publishMessage(topicName, data) {
  const dataBuffer = Buffer.from(data);
  pubsub
    .topic(topicName)
    .publisher()
    .publish(dataBuffer)
    .then(results => {
      const messageId = results[0];
      console.log(`Message ${messageId} published.`);
    })
    .catch(err => {
      console.error('ERROR:', err);
    });
}
```

**Comment by sadohert** [comment]
I just bumped into this too.  Following the docs here:

https://cloud.google.com/pubsub/docs/publisher#pubsub-publish-message-nodejs

I noticed the 2nd approach from @mooose was described.  When I looked at the github project these docs pointed to I was taken to:

https://github.com/googleapis/nodejs-pubsub/blob/master/samples/topics.js

... which is part of the googleapis project.

So seems like there is some redundancy happening, and the "GoogleCloudPlatform" project has out of date pubsub example code.
